src/components/AddProtocolModal/Alternative.jsx

```diff
@@ -0,0 +1,289 @@
+            const convertedValue = Number(value);
+
+            if (convertedValue < 1) {
+                return;
+            }
+
+            if (convertedValue >= data.maxRentingDurationUnits) {
+                return setData({
+                    ...data,
+                    minRentingDurationUnits: convertedValue,
+                    maxRentingDurationUnits: convertedValue + 1,
+                });
+            }
+        }
+
+        if (name === 'maxRentingDurationUnits') {
+            const convertedValue = Number(value);
+
+            if (convertedValue <= data.minRentingDurationUnits) {
+                return;
+            }
+        }
+
+        setData({
+            ...data,
+            [name]: typeof value === 'string' ? value.trim() : value,
+        });
+    };
+
+    const onStatusChange = (args) => {
+        console.log({ args });
+        const { status, data } = args;
+
+        if (status === 'error') {
+            console.error('Offer error', data);
+        }
+        if (status === 'seated') {
+            console.log('Transaction submitted:', data.txn);
+            console.log('Offer id:', data.offerId);
+        }
+        if (status === 'refunded') {
+            console.log('Offer refunded');
+        }
+        if (status === 'accepted') {
+            console.log('Offer accepted');
+        }
+
+        onModalClose();
+    };
+
+    const validate = () => {
+        const possibleErrors = {
+            utilityAmountIndex: checkNumber(data.utilityAmountIndex) ? '' : 'Required',
+            rentalFeePerUnitVal: checkNumber(data.rentalFeePerUnitVal) ? '' : 'Required',
+            collateralVal: checkNumber(data.collateralVal) ? '' : 'Required',
+            gracePeriodDuration: checkNumber(data.gracePeriodDuration) ? '' : 'Required',
+        };
+
+        console.log({ possibleErrors, utilityAmountIndex: data.utilityAmountIndex });
+
+        if (Object.values(possibleErrors).join('')) {
+            setErrors({ ...errors, ...possibleErrors });
+            return false;
+        }
+
+        return true;
+    };
+
+    const handleSubmit = () => {
+        console.log({ data, wallet });
+        if (!wallet || !validate()) throw new Error(`Not ready; wallet: ${wallet}, data: ${data}`);
+
+        const offerSpec = buildCreateRentalOfferSpec(data);
+        console.log({ offerSpec });
+        void wallet.makeOffer(
+            offerSpec.invitationSpec,
+            offerSpec.proposal,
+            offerSpec.offerArgs,
+            onStatusChange,
+            offerSpec.id,
+        );
+    };
+
+    console.log('utilityPurse: ', utilityPurse);
+
+    return (
+        open && (
+            <ModalWrapper className="add-protocol-modal">
+                {/*{submittedData && <RentalCreator data={submittedData} onSubmit={() => setSubmittedData(null)} />}*/}
+                <header>
+                    <h2 className="modal-title">Add Crabble Protocol</h2>
```

Comment:
Nor this component has the styling of `AddProtocolModal.jsx` @GrigoriGrigoryanBytepitch

Reply:
I did not add style for header, but now pushed. Can you please check it again ?

Reply:
Looks great man! I did some cleanup, merging now.